packages/mosaic/sql/src/functions/list.ts

```diff
@@ -0,0 +1,58 @@
+import type {ExprValue, MaybeArray} from '../types.js';
+import { asLiteral } from '../util/ast.js';
+import { argsList, fn } from '../util/function.js';
+import { ListNode } from "../ast/list.js";
+import {ExprNode} from "../ast/node.js";
+
+function listFn(name: string, expr: MaybeArray<ExprValue>, ...args: unknown[]) {
```

Comment:
I'm interpreting this as a helper for functions that take two lists as input. Instead of `expr` and `...args` I'd expect arguments like `list1` and `list2`, each receiving identical processing. I think the type signature for an input list should be something like `ExprValue | ExprValue[]`, supporting either a single expression or an array of list content. A string value could be mapped to a column reference. If array-typed, the array could be pre-processed (e.g., using `list`).